server/src/test/java/org/opensearch/common/cache/serializer/ICacheKeySerializerTests.java

```diff
@@ -43,10 +43,17 @@ public void testInvalidInput() throws Exception {
         ICacheKeySerializer<BytesReference> serializer = new ICacheKeySerializer<>(keySer);
 
         Random rand = Randomness.get();
-        byte[] randomInput = new byte[1000];
-        rand.nextBytes(randomInput);
-
-        assertThrows(OpenSearchException.class, () -> serializer.deserialize(randomInput));
+        // The first thing the serializer reads is a VInt for the number of dimensions.
+        // This is an invalid input for StreamInput.readVInt(), so we are guaranteed to have an exception
+        byte[] invalidVInt = { -1, -1, -1, -1, -1 };
+        byte[] randomBytes = new byte[1000];
+        rand.nextBytes(randomBytes);
+        // Concatenate the two byte[] into the invalid input
+        byte[] invalidInput = new byte[invalidVInt.length + randomBytes.length];
+        System.arraycopy(invalidVInt, 0, invalidInput, 0, invalidVInt.length);
+        System.arraycopy(randomBytes, 0, invalidInput, invalidVInt.length, randomBytes.length);
+
+        assertThrows(OpenSearchException.class, () -> serializer.deserialize(invalidInput));
```

Comment:
What if you simplified all of this to `assertThrows(OpenSearchException.class, () -> serializer.deserialize(new byte[] { 0x01 }));`? What benefit are you getting from the randomness given that the function will fail before reading the random values?

Reply:
True, we don't really need the rest of it. Made that change